kong/plugins/datadog/statsd_logger.lua

```diff
@@ -21,12 +21,12 @@ local statsd_mt = {}
 statsd_mt.__index = statsd_mt
 
 local env_datadog_agent_host = os.getenv 'KONG_DATADOG_AGENT_HOST'
-local env_datadog_agent_port = tonumber(os.getenv 'KONG_DATADOG_AGENT_PORT' or "")
+local env_datadog_agent_port = os.getenv 'KONG_DATADOG_AGENT_PORT'
 
 function statsd_mt:new(conf)
   local sock   = udp()
-  local host = conf.host or env_datadog_agent_host
-  local port = conf.port or env_datadog_agent_port
+  local host = env_datadog_agent_host or conf.host
```

Comment:
This makes the environment variable takes precedence over `kong.conf`, which has different behavior compared to what this originally intended.

Maybe we could remove the default value for `kong.conf`, and:

1. check if `conf.host` is set; use that if so;
2. check if `'KONG_DATADOG_AGENT_HOST'` env is set; use that if so;
3. use the default value

Reply:
Check https://github.com/Kong/kong/blob/4ac5e9123e14c9cdf8f5d792f97561cc2afee40b/kong/plugins/datadog/schema.lua#L76

removing those default values will get us a `nil` when the option is omitted.